modules/smartrtbBidAdapter.js

```diff
@@ -0,0 +1,105 @@
+import * as utils from 'src/utils';
+import {registerBidder} from 'src/adapters/bidderFactory';
+const BIDDER_CODE = 'smartrtb';
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['smrtb'],
+  isBidRequestValid: function(bid) {
+    return (bid.params.pubId !== null &&
+      bid.params.medId !== null &&
+      bid.params.zoneId !== null);
+  },
+  getDomain: function () {
+    if (!utils.inIframe()) {
+      return window.location.hostname
+    }
+    var origins = window.document.location.ancestorOrigins
+    if (origins && origins.length > 0) {
+      return origins[origins.length - 1]
+    }
+  },
+  buildRequests: function(validBidRequests, bidderRequest) {
+    let stack = (bidderRequest.refererInfo &&
+      bidderRequest.refererInfo.stack ? bidderRequest.refererInfo
+      : [])
+
+    const payload = {
+      start_time: utils.timestamp(),
+      tmax: 120,
+      language: window.navigator.userLanguage || window.navigator.language,
+      site: {
+        domain: this.getDomain(),
+        iframe: !bidderRequest.refererInfo.reachedTop,
+        url: stack && stack.length > 0 ? [stack.length - 1] : null,
+        https: (window.location.protocol === 'https:'),
+        referrer: bidderRequest.refererInfo.referer
+      },
+      imps: []
+    };
+
+    if (bidderRequest && bidderRequest.gdprConsent) {
+      payload.gdpr = {
+        applies: bidderRequest.gdprConsent.gdprApplies,
+        consent: bidderRequest.gdprConsent.consentString
+      };
+    }
+
+    for (let req of validBidRequests) {
+      payload.imps.push({
+        pub_id: req.params.pubId,
+        med_id: req.params.medId,
+        zone_id: req.params.zoneId,
+        bid_id: req.bidId,
+        imp_id: req.transactionId,
+        sizes: req.sizes,
+        force_bid: req.params.forceBid
+      });
+    }
+
+    return {
+      method: 'POST',
+      url: '//pubs.smrtb.com/json/publisher/prebid',
+      data: JSON.stringify(payload)
+    };
+  },
+  interpretResponse: function(serverResponse, bidRequest) {
+    const bidResponses = [];
+    let res = serverResponse.body;
+    if (!res.bids || !res.bids.length) {
+      return []
+    }
+
+    for (let bid of serverResponse.body.bids) {
```

Comment:
The `for...of` comment above applies here too.

Reply:
Fixed.